### PR TITLE
Don't plot graph past final cycle point.

### LINKED ISSUE #1650: cylc graph - don't plot nodes beyond the final cycle point.
```
[cylc]
    cycle point format = %Y-%m-%d
[scheduling]
    initial cycle point = 2015-01-01
    final cycle point   = 2015-01-03
    [[dependencies]]
      [[[R1]]]
         graph = start => foo
      [[[P1D]]]
         graph = foo[-P1D] => foo => bar & baz
      [[[R1/P0D]]]
        graph = baz => stop
[visualization]
    number of cycle points = 4
    [[node attributes]]
        start = "style=filled", "fillcolor=slategray"
        stop = "style=filled", "fillcolor=red"
```
foo

### PR DESCRIPTION
**Pull request by hjoliver**
Close #1650.

### CODE REVIEW

**Comment by hjoliver** [comment]
(not ready yet - this fixes the problem, but raises other questions...)

**Comment by hjoliver** [comment]
(TODO - a couple of graph comparison tests).

File: bin/cylc-graph [hunk 1]
```diff
@@ -150,8 +150,10 @@ def main():
         __doc__, jset=True, prep=True,
         argdoc=[
             ('[SUITE]', 'Suite name or path'),
-            ('[START]', 'Initial cycle point to plot (default=2999010100)'),
-            ('[STOP]', 'Final cycle point to plot (default=2999010123)')])
```
Comment (hjoliver) [inline bin/cylc-graph]: Outdated documentation (these defaults pre-date cylc-6).

File: lib/cylc/config.py [hunk 2]
```diff
@@ -1880,10 +1884,12 @@ def get_graph_raw(self, start_point_string, stop_point_string,
                 if stop_point is not None and point > stop_point:
                     # Beyond requested final cycle point.
                     break
+                if suite_final_point is not None and point > suite_final_point:
+                    # Beyond suite final cycle point.
+                    break
                 if stop_point is None and len(new_points) > n_points:
                     # Take n_points cycles from each sequence.
                     break
-                not_initial_cycle = (point != i_point)
```
Comment (hjoliver) [inline lib/cylc/config.py]: Not used anywhere.

File: lib/cylc/config.py [hunk 3]
```diff
@@ -2035,8 +2041,6 @@ def load_graph(self):
             )
         back_comp_initial_tasks = list(start_up_tasks)
 
-        has_non_async_graphs = False
-
```
Comment (hjoliver) [inline lib/cylc/config.py]: Not used anywhere.

**Comment by hjoliver** [comment]
@cylc/core - the delay with this pull request is that it (naturally) stops the gcylc graph view from plotting nodes past the final cycle point.  Although this is the correct thing to do, it is inconsistent with the other views, which show the final task proxies held at their "next point" beyond the find point.  This means of stopping a suite dates back to before cylc-6, when the final cycle point was nothing but an arbitrary point on a cycling sequence (nothing special could happen there) and it was possible to restart a finished suite and carry on as normal after changing the final point and releasing the held tasks.  Now that we can have special behaviour anywhere, in particular at the final cycle point, it doesn't make much sense to allow task proxies to spawn past the final point, we should stop doing that.  Instead, the final tasks should just disappear once finished and no longer needed, or perhaps go into a final "finished" state.  This makes the issue a bit less trivial though.  I'm inclined to put this in now, because it does fix a confusing aspect of static graph visualization, and just accept that the graph view will not show the final task proxies that are held beyond the final cycle point.  Do you agree?  If so, I'll update the graph comparison tests and so we can merge this.  Otherwise it'll have to stay in "soon" until the wider issue is addressed...

**Comment by benfitzpatrick** [comment]
Looks fine to me

**Comment by matthewrmshin** [comment]
@hjoliver I have no strong opinion on this, so please go ahead.

**Comment by arjclark** [comment]
My only query would be, does this have any implications for extending a suite run? Or is it purely a visualisation thing?

**Comment by hjoliver** [comment]
It's purely visualisation.

**Comment by hjoliver** [comment]
New tests added.  All tests pass in my environment.

@matthewrmshin - please assign reviews (may be done already?).

**Comment by matthewrmshin** [comment]
@arjclark please review 2.

**Comment by arjclark** [comment]
Looks ok to me. Test battery passing in my environment. Manual testing of cylc graph with a simple suite seems to show new behaviour working correctly (and as would be expected in a suite with a defined final cycle point).

**Pull request merged**